### ISSUE TITLE
Check that every function has a memory safety attribute (e.g., @safe)

### ISSUE THREAD

**Issue by mcandre** [dlang-community/D-Scanner#715]
As an engineer, I want my D code to promote better memory safety, so that my applications run more reliably. Could D-Scanner add a rule to check that every function has an associated memory safety attribute (either `@safe`, or `@trusted`, or `@system`)?

**Comment by BBasile** [comment]
Yes that's possible. However since D functions are often templatized and since their attributes are inferred this brings a question: should the the check  warn on templatized function without memory safety ? 
(I think it shouldn't).

**Comment by wilzbach** [comment]
There's one very similar one which enforces this for all unittests (it's used for Phobos)
